karapace/utils.py

```diff
@@ -191,6 +194,27 @@ def convert_to_int(object_: dict, key: str, content_type: str):
 
 
 class KarapaceKafkaClient(KafkaClient):
+    def __init__(self, **configs):
+        kafka.client_async.BrokerConnection = KarpaceBrokerConnection
```

Comment:
can this be done without monkey patching?

Reply:
yes it can, but it will require copy pasting a very large method, so i figured this is the lesser of 2 evils